### ISSUE TITLE
[cssom] getComputedStyle for non-existent pseudo-element

### ISSUE THREAD

**Issue by zcorpan** [w3c/csswg-drafts#1761]
https://drafts.csswg.org/cssom/#resolved-values

> width
> ...
> If the property applies to the element or pseudo\-element and the resolved value of the display property is not none or contents, then the resolved value is the used value\. Otherwise the resolved value is the computed value\.

See https://github.com/w3c/web-platform-tests/pull/7020

What should happen for `getComputedStyle(el, '::before').width` if el has no `::before` box generated? This appears to not be particularly well defined.

https://drafts.csswg.org/cssom/#terminology says

> In this specification the ::before and ::after pseudo-elements are assumed to exist for all elements even if no box is generated for them.

**Comment by Loirooriol** [comment]
I would expect the same behavior as when an element does not generate boxes due to `display: none` or `display: contents`. In fact the spec is also missing the case of an element whose `display` is not `none` or `contents` but no box is generated due to a `display: none` ancestor. Then, all Firefox, Chrome and Edge return the computed value, but according to the spec it seems it should be the used value.

**Comment by tabatkins** [comment]
Yup, it should be the same as a `display:none` element.  (In other words, the pseudo-element *always* exists, but most of the time just doesn't generate a box due to `content`.)

**Comment by FremyCompany** [comment]
Could probably be merged with #2149
